R/data-repository.R

```diff
@@ -56,12 +58,60 @@ DataRepository <- R6::R6Class(
     }
   ),
   public = list(
+    #' @description
+    #' Adds a column to the data repository
+    #' @param column Column to add
+    addColumn = function(column){
+      validateIsOfType(column, DataColumn)
+      rClr::clrCall(self$ref, "Add", column$ref)
+      #we need to reset the cache when adding a new column
+      private$.columns <- NULL
+      private$.baseGrid <- NULL
+    },
+    #' @description
+    #' Initialize a new instance of the class
+    #' @param ref Optional underlying DataRepository. If it is not provided, a new instance will be created
+    #' @return A new `DataRepository` object.
+    initialize = function(ref = NULL) {
+      super$initialize(ref %||%  rClr::clrNew("OSPSuite.Core.Domain.Data.DataRepository"))
+    },
     #' @description
     #' Print the object to the console
     #' @param ... Rest arguments.
     print = function(...) {
       private$printClass()
       invisible(self)
+    },
+    #' @description
+    #' Adds a new entry to meta data list or changes its value if the name is already present.
+    #' If \code{value} is \code{NULL}, the entry with corresponding name is deleted from meta data set.
+    #'
+    #' @param name Name of new meta data list entry
+    #' @param value Value of new meta data list entry
+    addMetaData = function(name, value) {
+      if (length(name) != 1) {
+        stop(messages$errorMultipleMetaDataEntries())
+      }
+      validateIsString(name)
+      validateIsString(value)
+      dataRepositoryTask <- getNetTask("DataRepositoryTask")
+      rClr::clrCall(dataRepositoryTask, "AddMetaData", self$ref, name, value)

```

Comment:
is there a way to coherce a parameter to string in R. If yes, then we can remove the need for value to be a string

Reply:
`as.character()` works e.g. for numerics but not for class objects, so maybe leave the check here.